apps/dashboard/src/components/common/SideBarAccordion.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useCallback, useState } from "react";
+import { ChevronDownIcon } from "@radix-ui/react-icons";
+import cn from "classnames";
+import { NavAccordionItem } from "@/types/navbar";
+import SideBarItem from "./SideBarItem";
+import { Void } from "@litespace/types";
+
+export const SideBarAccordion: React.FC<{
+  item: NavAccordionItem;
+  open?: NavAccordionItem["id"];
+  onClick: Void;
+}> = ({ item, onClick }) => {
+  const [open, setOpen] = useState<NavAccordionItem["id"] | null>(null);
+
+  const show = useCallback((id: NavAccordionItem["id"]) => setOpen(id), []);
+  const hide = useCallback(() => setOpen(null), []);
+  const toggle = useCallback(
+    (id: NavAccordionItem["id"]) => {
+      if (open === id) return hide();
+      return show(id);
+    },
+    [hide, open, show]
+  );
+
+  return (
+    <div className="tw-w-full">
+      <ul
+        className={`tw-w-full ${
+          open ? "tw-border tw-border-border tw-rounded" : ""
+        }`}
+      >
+        <li key={item.id}>
+          <div
+            onClick={() => toggle(item.id)}
+            className="tw-flex tw-items-center tw-justify-between tw-h-[45px] tw-px-4 tw-py-2 tw-border-b tw-border-border hover:tw-bg-selection tw-cursor-pointer"
+          >
+            <div className="text-xl">{item.title}</div>
+            <div
+              data-open={item.id === open}
+              className={cn(
+                "data-[open=true]:tw-rotate-180 tw-transition-transform tw-duration-300"
+              )}
+            >
+              <ChevronDownIcon />
+            </div>
+          </div>
+          <div
+            data-open={item.id === open}
+            className={cn(
+              "tw-text-foreground-light",
+              "data-[open=true]:tw-border-b data-[open=true]:tw-border-border-strong",
+              "tw-grid tw-grid-rows-[0fr] data-[open=true]:tw-grid-rows-[1fr]",
+              "tw-transition-[grid-template-rows] tw-duration-200 tw-ease-in tw-px-4 data-[open=true]:tw-py-2"
```

Comment:
Are you sure that this the right way to do it? Because the `tw-` prefix doesn't work on the dashboard. @mostafakamar2308

Reply:
@neuodev 
I didn't see any problems but I removed it to make the style consistent with the dashboard itself